aip/aog/3010.md

```diff
@@ -0,0 +1,55 @@
+---
+aip:
+  id: 3010
+  state: draft
+  created: 2020-01-14
+  updated: 2020-01-14
+  type: guidance
+  scope: aog
+js:
+  - /assets/js/graphviz/viz.js
+  - /assets/js/graphviz/lite.render.js
+  - /assets/js/aip/aip-graphviz.js
+permalink: /aog/3010
+---
+
+# Actions on Google Vertical Integration Webhook Format
+
+If an action is part of a vertical program that uses Google Fulfillment, it
+usually needs to expose a webhook that Google will call to execute user queries
+via a structured protocol. This AIP defines requirement for the webhook format.
+
+## Guidance
+
+### Base URL in Action Package
+
+The `baseUrl` configured in Action Package defines the common part of the
+actual execution URL of the webhook. The base URL **may** include a trailing
+slash, but **must** work if `:customMethodName` is appended directly. The
+endpoint **must** be HTTPS and **must** have valid certificate.
+
+Examples:
+
+- `https://us-central1.cloudfunctions.net/myWebhook/`
+- `https://test.com/assistant/api`
+
+### Actual execution URL
+
+The actual execution URL **should** be `baseUrl` appended with
+`:customMethodName`, where `customMethodName` is the Google Fulfillment method
+name.
+
+Examples:
+
+- `https://us-central1.cloudfunctions.net/myWebhook/:completeTransaction`
+- `https://test.com/assistant/api:completeTransaction`
+
+### Google Fulfillment method name
```

Comment:
I think maybe this section should be higher up, and maybe give a sentence or two of context about how GF uses custom methods, since the rest of the AIP refers to custom methods without having explained them.

Reply:
Should we even have this section in this doc? The first part's audience is external developers where as the method name is defined by Google.

Reply:
I think we should mention it, because 3P developers needs to implement it accordingly.

Reply:
I agree we should mention it.  I think most AIPs have both internal and external audiences, and the internal API author audience is perhaps even the more important one.

Reply:
I moved this section up and renamed it. (didn't delete it)

Reply:
Sorry, totally missed that!